src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -37,70 +40,63 @@ object VariantSampleMatrix {
     if (!hadoopExists(hConf, dirname))
       fatal(s"no VDS found at `$dirname'")
 
-    val vaSchema = dirname + "/va.schema"
-    val saSchema = dirname + "/sa.schema"
-    val globalSchema = dirname + "/global.schema"
+    val metadataFile = dirname + "/metadata.json.gz"
     val pqtSuccess = dirname + "/rdd.parquet/_SUCCESS"
-    val metadataFile = dirname + "/metadata.ser"
 
     if (!hadoopExists(hConf, metadataFile))
-      fatal("corrupt VDS: no metadata.ser file.  Recreate VDS.")
-
-    val (sampleIds, sampleAnnotations, globalAnnotation, wasSplit) =
-      readDataFile(dirname + "/metadata.ser", hConf) { dis =>
-        try {
-          val serializer = SparkEnv.get.serializer.newInstance()
-          val ds = serializer.deserializeStream(dis)
-
-          val m = ds.readObject[Int]
-          if (m != magicNumber)
-            fatal("Invalid VDS: invalid magic number")
-
-          val v = ds.readObject[Int]
-          if (v != fileVersion)
-            fatal("Old VDS version found")
-
-          val sampleIds = ds.readObject[IndexedSeq[String]]
-          val sampleAnnotations = ds.readObject[IndexedSeq[Annotation]]
-          val globalAnnotation = ds.readObject[Annotation]
-          val wasSplit = ds.readObject[Boolean]
+      fatal("corrupt or old VDS: no `metadata.json.gz' file found.\n  Recreate VDS with current version of Hail.")
+
+    val json = try {
+      readFile(metadataFile, hConf)(
+        in => JsonMethods.parse(Source.fromInputStream(in).mkString)
+      )
+    } catch {
+      case e: Throwable => fatal(s"corrupt VDS: invalid metadata file: ${e.getMessage}.\n  " +
+        s"Recreate VDS with current version of Hail")
+    }
 
-          ds.close()
-          (sampleIds, sampleAnnotations, globalAnnotation, wasSplit)
+    val fields = json match {
+      case jo: JObject => jo.obj.toMap
+      case _ => fatal("corrupt VDS: invalid metadata json.\n  Recreate VDS with current version of Hail.")
+    }
 
-        } catch {
-          case e: Exception =>
-            println(e)
-            fatal(s"Invalid VDS: ${e.getMessage}\n  Recreate with current version of Hail.")
-        }
+    def getAndCastJSON[T <: JValue](fname: String, expected: String)(implicit tct: ClassTag[T]): T =
```

Comment:
Don't pass expected, use `tct.runtimeClass.getName` to match `other.getClass.getName`.